### PR TITLE
core/muxer: Deprecate `StreamMuxerExt::next_{inbound,outbound}`

### PR DESCRIPTION
**Pull request by thomaseizinger**
# Description

## Links to any relevant issues

- https://github.com/libp2p/rust-libp2p/pull/2952

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: core/CHANGELOG.md [hunk 1]
```diff
@@ -5,8 +5,11 @@
 - Remove default features. If you previously depended on `secp256k1` or `ecdsa` you need to enable these explicitly 
   now. See [PR 2918].
 
+- Deprecate `StreamMuxerExt::next_{inbound,outbound}`. See [PR XXXX].
+
 [PR 2915]: https://github.com/libp2p/rust-libp2p/pull/2915
 [PR 2918]: https://github.com/libp2p/rust-libp2p/pull/2918
+[PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX
```
Comment (thomaseizinger) [inline core/CHANGELOG.md]:
```suggestion
- Deprecate `StreamMuxerExt::next_{inbound,outbound}`. See [PR 3002].

[PR 2915]: https://github.com/libp2p/rust-libp2p/pull/2915
[PR 2918]: https://github.com/libp2p/rust-libp2p/pull/2918
[PR 3002]: https://github.com/libp2p/rust-libp2p/pull/3002
```

File: muxers/mplex/tests/two_peers.rs [hunk 2]
```diff
@@ -174,7 +187,10 @@ fn protocol_not_match() {
             .await
             .unwrap();
 
-        let mut outbound = client.next_outbound().await.unwrap();
+        // Just calling `poll_outbound` without `poll` is fine here because mplex makes progress through all `poll_` functions. It is hacky though.
+        let mut outbound = poll_fn(|cx| client.poll_outbound_unpin(cx))
+            .await
+            .expect("unexpected error");
```
Comment (thomaseizinger) [inline muxers/mplex/tests/two_peers.rs:193]: Ugly repetition but will go away with: https://github.com/libp2p/rust-libp2p/pull/2952

**Review by mxinden — APPROVED** [review]
Thanks!

**Comment by mxinden** [comment]
@thomaseizinger with `v0.49.0` released, can you create a pull request removing the deprecated methods?

**Comment by thomaseizinger** [comment]
> @thomaseizinger with `v0.49.0` released, can you create a pull request removing the deprecated methods?

Working on it here: https://github.com/libp2p/rust-libp2p/pull/3031

**Pull request merged**